pincer/objects/guild/guild.py

```diff
@@ -1236,6 +1237,165 @@ async def modify_user_voice_state(
+            The ID of the emoji
+
+        Returns
+        -------
+        :class:`~pincer.objects.guild.emoji.Emoji`
+            The emoji object.
+        """
+        return Emoji.from_dict(
+            construct_client_dict(
+                self._client,
+                await self._http.get(f"guilds/{self.id}/emojis/{id}")
+            )
+        )
+
+    async def create_emoji(
+        self,
+        *,
+        name: str,
+        image: str,
+        roles: List[Snowflake],
+        reason: Optional[str] = None
+    ) -> Emoji:
+        """|coro|
+        Creates a new emoji for the guild.
+        Requires the ``MANAGE_EMOJIS_AND_STICKERS`` permission.
+
+        Emojis and animated emojis have a maximum file size of 256kb.
+        Attempting to upload an emoji larger than this limit will fail.
+
+        Parameters
+        ----------
+        name : :class:`str`
+            Name of the emoji
+        image : :class:`str`
+            The 128x128 emoji image data
+        roles : List[:class:`~pincer.utils.snowflake.Snowflake`]
+            Roles allowed to use this emoji
+        reason : Optional[:class:`str`]
+            The reason for creating the emoji |default| :data:`None`
+
+        Returns
+        -------
+        :class:`~pincer.objects.guild.emoji.Emoji`
+            The newly created emoji object.
+        """
+        data = await self._http.post(
+            f"guilds/{self.id}/emojis",
+            data={
+                "name": name,
+                "image": image,
+                "roles": roles
+            },
+            headers=remove_none({"X-Audit-Log-Reason": reason})
+        )
+        return Emoji.from_dict(
+            construct_client_dict(self._client, data)
+        )
+
+    async def edit_emoji(
+        self,
+        id: Snowflake,
+        *,
+        name: Optional[str] = None,
+        roles: Optional[List[Snowflake]] = None,
+        reason: Optional[str] = None
+    ) -> Emoji:
+        """|coro|
+        Modifies the given emoji.
+        Requires the ``MANAGE_EMOJIS_AND_STICKERS`` permission.
+
+        Parameters
+        ----------
+        id : :class:`~pincer.utils.snowflake.Snowflake`
+            The ID of the emoji
+        name : Optional[:class:`str`]
+            Name of the emoji |default| :data:`None`
+        roles : Optional[List[:class:`~pincer.utils.snowflake.Snowflake`]]
+            Roles allowed to use this emoji |default| :data:`None`
+        reason : Optional[:class:`str`]
+            The reason for editing the emoji |default| :data:`None`
+
+        Returns
+        -------
+        :class:`~pincer.objects.guild.emoji.Emoji`
+            The modified emoji object.
+        """
+        data = await self._http.patch(
+            f"guilds/{self.id}/emojis/{id}",
+            data={
+                "name": name,
+                "roles": roles
+            },
+            headers=remove_none({"X-Audit-Log-Reason": reason})
```

Comment:
why not just moving `remove_none` inside of the `__send` method?

```diff
        if isinstance(data, dict):
-           data = dumps(data)
+           if (data := remove_none(data)):
+               data = dumps(data)
```

Reply:
I'd keep it like this since the function might be useful in other cases in the future

Reply:
Imo no walrus operator is better